LiteCore/Query/QueryParser.cc

```diff
@@ -44,6 +45,38 @@ namespace litecore {
 #pragma mark - UTILITY FUNCTIONS:
 
     namespace qp {
+        struct caseInsensitiveSlice {
+            size_t operator() (pure_slice const& s) const {
+                uint32_t h = 2166136261;
+                for (size_t i = 0; i < s.size; i++) {
+                    h = (h ^ tolower(s[i])) * 16777619;
+	            }
+                        
+	            return h;
+            }
+
+            bool operator()(const slice& a, const slice& b) const {
+                return a.caseEquivalent(b);
+            }
+        };
+
+        using case_insensitive_set = std::unordered_set<slice, caseInsensitiveSlice, caseInsensitiveSlice>;
+
+        bool isImplicitBool(const Value* op) {
+            if(!op) {
+                return false;
+            }
+
+            static const case_insensitive_set implicitBoolOps = {
+                "!="_sl, "="_sl, ">"_sl, "<"_sl, ">="_sl, "<="_sl,
+                "IS"_sl, "IS NOT"_sl, "NOT"_sl,
+                "BETWEEN"_sl, "AND"_sl, "OR"_sl,
+                "NOT IN"_sl, "EVERY"_sl, "ANY AND EVERY"_sl
```

Comment:
* We also support SOME which is the same as ANY. 
* How about LIKE?

Reply:
LIKE actually translates to its own function that already returns a bool.  Not sure about SOME, I don't see it being mapped by the query parser to anything  I don't think it works right now.